modules/ximgproc/src/structured_edge_detection.cpp

```diff
@@ -243,7 +245,7 @@ static void gradientHist(const cv::Mat &src, cv::Mat &magnitude, cv::Mat &histog
         float *pHist = histogram.ptr<float>(i/pSize);
 
         for (int j = 0; j < phase.cols; ++j)
-            pHist[cvRound((j/pSize + pPhase[j])*nBins)] += pMagn[j] / CV_SQR(pSize);
+            pHist[cvRound((j/pSize + pPhase[j]*phaseRoundingCoeff - phaseRoundingShift)*nBins)] += pMagn[j] / CV_SQR(pSize);
```

Comment:
We can use something like this in case of occasional out of range results in floating point calculations:
```
size_t index  = cvRound(... accurate formula without magic coefficients ...)
index = max(0, min(index, <sizeof pHist> - 1));
pHist[index] = ...
```